src/Flow/Mongo/MongoFile.php

```diff
@@ -90,12 +90,22 @@ public function saveChunk()
             ) {
                 throw new \Exception("Invalid upload! (size: {$actualChunkSize})");
             }
-            $chunk['data'] = new \MongoBinData($data);
+            $chunk['data'] = new \MongoBinData($data, 0); // \MongoBinData::GENERIC is not defined for older mongo drivers
             $this->config->getGridFs()->chunks->findAndModify($chunkQuery, $chunk, [], ['upsert' => true]);
             unlink($file['tmp_name']);
 
+            $this->ensureIndices();
+
             return true;
         } catch (\Exception $e) {
+            error_log("Could not store chunk: " . $e->getMessage() . "\n" . $e->getTraceAsString());
+            try {
+                if (isset($chunkQuery)) {
+                    $this->config->getGridFs()->chunks->remove($chunkQuery);
+                }
+            } catch (\Exception $e2) {
```

Comment:
Why I catch the base exception here: Whatever went wrong inserting the chunk might as well fail when we try to remove the chunk - hence we catch the same exception to assure the method signature (returning true/false, but no known exceptions). Does that make sense to you?